### PR TITLE
examples: Update openai chat example

### PR DESCRIPTION
**Pull request by tmc**
This updates the openai chat example to use llms.Model.GenerateContent.

### CODE REVIEW

File: examples/openai-chat-example/openai_chat_example.go [hunk 1]
```diff
@@ -11,22 +11,33 @@ import (
...
 		log.Fatal(err)
 	}
 	ctx := context.Background()
-	completion, err := llm.Call(ctx, []schema.ChatMessage{
-		schema.SystemChatMessage{Content: "Hello, I am a friendly chatbot. I love to talk about movies, books and music. Answer in long form yaml."},
-		schema.HumanChatMessage{Content: "What would be a good company name a company that makes colorful socks?"},
+	_, err = llm.GenerateContent(ctx, []llms.MessageContent{
+		{
+			Role: schema.ChatMessageTypeSystem,
+			Parts: []llms.ContentPart{
```
[the remark this answers is not in the document]
Comment (tmc) [reply examples/openai-chat-example/openai_chat_example.go:22]: Thanks for the api suggestions, let me mull these over a bit.

[the remark this answers is not in the document]
Comment (tmc) [reply examples/openai-chat-example/openai_chat_example.go:22]: Added a small PR to add a helper in https://github.com/tmc/langchaingo/pull/549

**Comment by eliben** [comment]
LMK if you're going to update this PR now that the helpers landed, or if you want my help updating all the examples!

I'll send a PR to update the vertex example and add a googleai example

**Comment by tmc** [comment]
Please feel free to open PRs to update the other examples, otherwise I can probably get to it later today.

**Comment by eliben** [comment]
> Please feel free to open PRs to update the other examples, otherwise I can probably get to it later today.

Since the examples don't build without a `go.work` file anyway and running `go mod tidy` in CI fails, I would recommend tagging a new version of `langchaingo` first, and then updating all the examples once to use the newest released version. Otherwise we'll do a lot of busy work updating them now to work with `go.work` and then again after the tagging to make sure all the `go.mod / go.sum` files are updated and everything builds properly in CI.

**Comment by eliben** [comment]
I went ahead and updated the OpenAI examples here: https://github.com/tmc/langchaingo/pull/573

(figured these will be the most likely to be looked at).

There's still leftover work to do on these after the new version is tagged, so I'd really prefer to only do one examples update - but #573 is a step in the right direction!

**Comment by tmc** [comment]
@eliben I went ahead and pushed a v0.1.4-alpha.0 tag that can be targetted here, I figure we can get the examples in order and then barring any issues get v0.1.4 released then point the examples at the release tag.

**Comment by tmc** [comment]
Subsumed by #573

**Comment by eliben** [comment]
@tmc with #576 all examples have now been updated to the new APIs and work with the alpha tag you created. I think it's safe to tag v0.1.4 itself now

Once you do, I can update the examples to point to the final tag.

**Pull request closed without merging**